crate/operator/utils/kopf.py

```diff
@@ -83,17 +86,21 @@ def __init__(
         :param run_on_dep_failures: whether we should still execute if our dependencies
             have failed. This is useful for handlers that clean up resources after
             other handlers have finished, and always need to run.
+        :param timeout: the overall runtime of the handler
         """
         self.namespace = namespace
         self.name = name
         self.ref = ref
         self._context = context
         self.depends_on = depends_on if depends_on is not None else []
         self.run_on_dep_failures = run_on_dep_failures
+        self.timeout = timeout
 
     def __call__(self, **kwargs: Any):
         return functools.partial(self._subhandler, **kwargs)
 
+    @crate.on.error(error_handler=crate.send_error_notification)
```

Comment:
Should we put these here or on specific `handle()` methods in each handler? Then you can fine-tune it a bit more and send specific failure notifications for specific things that failed?...